ios/Podfile

```diff
@@ -87,4 +87,50 @@ post_install do |installer|
       config.build_settings['ENABLE_BITCODE'] = 'NO'
     end
   end
-end
+
+  installer.pods_project.targets.each do |target|
+    target.build_configurations.each do |config|
+
+      # You can remove unused permissions here
+      # for more infomation: https://github.com/BaseflowIT/flutter-permission-handler/blob/develop/permission_handler/ios/Classes/PermissionHandlerEnums.h
+      # e.g. when you don't need camera permission, just add 'PERMISSION_CAMERA=0'
+      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
+        '$(inherited)',
+
+        ## dart: PermissionGroup.calendar
+        'PERMISSION_EVENTS=0',
+
+        ## dart: PermissionGroup.reminders
+        'PERMISSION_REMINDERS=0',
+
+        ## dart: PermissionGroup.contacts
+        'PERMISSION_CONTACTS=0',
+
+        ## dart: PermissionGroup.camera
+        # 'PERMISSION_CAMERA=0',
+
+        ## dart: PermissionGroup.microphone
+        'PERMISSION_MICROPHONE=0',
+
+        ## dart: PermissionGroup.speech
+        'PERMISSION_SPEECH_RECOGNIZER=0',
+
+        ## dart: PermissionGroup.photos
+        'PERMISSION_PHOTOS=0',
+
+        ## dart: [PermissionGroup.location, PermissionGroup.locationAlways, PermissionGroup.locationWhenInUse]
+        'PERMISSION_LOCATION=0',
+
+        ## dart: PermissionGroup.notification
+        'PERMISSION_NOTIFICATIONS=0',
+
+        ## dart: PermissionGroup.mediaLibrary
+        'PERMISSION_MEDIA_LIBRARY=0',
+
+        ## dart: PermissionGroup.sensors
+        'PERMISSION_SENSORS=0'
+      ]
+
+    end
+  end
+end
```

Comment:
I am not sure about these permissions - some questions

1. Are we sure we're not removing existing permissions?
2. For new permissions in iOS, can we make sure that there is a "user visible string" for them - because without a user visible string, app will get rejected from the app store. 

This is why I was managing iOS permissions inside the iOS project so far. But if this works, cool.

Reply:
This is purely from the doc for setting up permission_handler: https://pub.dev/packages/permission_handler
This should fix the problem permission_handler had in earlier versions when the app was registered as handling all permissions just because the permission_handler plugin had support for them regardless of if the actual app used them or not. 

Can't say I know this pod scripting so I hope it does what it says.

Reply:
thanks. let's see.